Azure/azure-service-operator
# Rotate SQL User credentials

Closes #537 

**What this PR does / why we need it**:
Currently there is a way to roll admin credentials to the SQL server, but not SQL Database user credentials. This extends SQL Action to add that functionality.

When checking if the user exists or not, the code was using the kube namespaced name exclusively, thus it would always come up as a negative if the secret client was Key Vault. Refactored logic around with `dbUserCustomNamespace`, put the check for the sql user secret key value into one function.

**Special notes for your reviewer**:
To test:
- Create SQL Server + DB
- Create Firewall rule
- Create Create SQL User
- Create SQL action with action `rollusercreds`. Specify SQL User from above. Observe in Key Vault that credentials have updated.

**How does this PR make you feel**:
![gif](https://media.giphy.com/media/T8pgGGblQ9Zks/giphy.gif)

**If applicable**:
